Unable to compile diesel v1.1.1

<!--
If you want to report a bug, we added some points below you can fill out. If you want to request a feature, feel free to remove all the irrelevant text. In addition to this [issue tracker](https://github.com/diesel-rs/diesel/issues), you can also talk to Diesel maintainers and users on [Gitter](https://gitter.im/diesel-rs/diesel).
-->

## Setup

### Versions

- **Rust:** Nightly-2018-01-20
- **Diesel:** v1.1.1
- **Database:** Postgres
- **Operating System** Linux

### Feature Flags

- **diesel:** postgres, chrono, r2d2

## Problem Description
Unable to compile - output when compiling is:
```
   Compiling diesel v1.1.1
error[E0603]: trait `Error` is private
  --> /home/am0d/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel-1.1.1/src/r2d2.rs:11:22
   |
11 | pub type PoolError = self::r2d2::Error;
   |                      ^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```

### What are you trying to accomplish?
Compile app.

### What is the expected output?
No errors.

### What is the actual output?
See above.

### Are you seeing any additional errors?
No.

### Steps to reproduce
Attempt to compile an application with a dependency on diesel v1.1.1 with the r2d2 feature enabled.
<!--
Please include as much of your codebase as needed to reproduce the error.  If the relevant files are large, please consider linking to a public repository or a [Gist](https://gist.github.com/).

Please post as much of your database schema as necessary. If you are using `infer_schema!`, you can use `diesel print-schema` and post the relevant parts from that.
-->

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.


<!--
Thank you for your submission!  You're helping make Diesel more robust 🎉

We'll try to respond as quickly as possible.
-->

--- a/diesel/Cargo.toml
+++ b/diesel/Cargo.toml
@@ -31,7 +31,7 @@ num-traits = { version = "0.1.35", optional = true }
 num-integer = { version = "0.1.32", optional = true }
 bigdecimal = { version = "0.0.10", optional = true }
 bitflags = { version = "1.0", optional = true }
-r2d2 = { version = ">= 0.7, < 0.9", optional = true }
+r2d2 = { version = ">= 0.8, < 0.9", optional = true }
 
 [dev-dependencies]
 cfg-if = "0.1.0"

